src/DataType.cpp

```diff
@@ -42,6 +44,35 @@ std::string data_type_to_string(DataType dtype) {
     return str;
 }
 
+DataType string_to_data_type(std::string dtype) {
+
+    struct data_type_map : public std::map<std::string, DataType> {
```

Comment:
Don't inherit from STL containers - especially when this is not necessary.

I would rather use something like this (if you really want to use a map):
```c++ 
static std::map<std::string, DataType> type_map = {
    {"bool", DataType::Bool},
    {"char", DataType::Char},
    ...
}
```